# **Identifikasi pada data tersebut apakah terdapat perbedaan karakteristik data train pada praktikum sebelumnya.**

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
test = pd.read_csv('/content/test_features.csv')
train = pd.read_csv('/content/train_features.csv')

In [6]:
test

ID  tahun_kelahiran pendidikan status_pernikahan   pendapatan  \
0     2241             1957    Sarjana               NaN  120660151.0   
1     2274             1968     Doktor           Menikah  163551821.0   
2     1107             1968        SMA           Menikah   29857000.0   
3     4478             1971     Doktor           Menikah  117949098.0   
4     5080             1974    Sarjana   Rencana Menikah  164761134.0   
...    ...              ...        ...               ...          ...   
3813   271             1984   Magister           Sendiri   66320000.0   
3814  3334             1972    Sarjana           Menikah  159651946.0   
3815  3478             1968   Magister   Rencana Menikah   84058923.0   
3816    89             1966    Sarjana   Rencana Menikah  106301000.0   
3817  2151             1973    Sarjana           Menikah  140353000.0   

      jumlah_anak_balita  jumlah_anak_remaja  terakhir_belanja  belanja_buah  \
0                    0.0                 1.0              63.0      122277.0   
1                    0.0                 1.0              58.0       35761.0   
2                    0.0                 0.0              34.0        8092.0   
3                    0.0                 1.0              82.0        4872.0   
4                    0.0                 0.0              28.0      343208.0   
...                  ...                 ...               ...           ...   
3813                 1.0                 0.0              19.0       12138.0   
3814                 0.0                 0.0              90.0       32119.0   
3815                 0.0                 1.0              46.0        8198.0   
3816                 0.0                 1.0              74.0       16184.0   
3817                 0.0                 0.0               9.0       62713.0   

      belanja_daging  belanja_ikan  belanja_kue  pembelian_diskon  \
0           541399.0      214192.0      84305.0               4.0   
1           353335.0       63365.0      41112.0               2.0   
2            22253.0       30345.0      26299.0               2.0   
3           126061.0           0.0       9745.0               3.0   
4          1416462.0      236196.0     107776.0               0.0   
...              ...           ...          ...               ...   
3813         38437.0           0.0      10115.0               4.0   
3814       1032587.0      176078.0      78693.0               0.0   
3815         47595.0       11392.0      10540.0               NaN   
3816         78897.0           NaN          0.0               3.0   
3817       1707412.0      453152.0          NaN               2.0   

      pembelian_web  pembelian_toko  keluhan tanggal_menjadi_anggota  
0               6.0            10.0      0.0                     NaN  
1               5.0            10.0      0.0                     NaN  
2               0.0             5.0      0.0              2013-08-06  
3               5.0             7.0      0.0                     NaN  
4               1.0             8.0      0.0                     NaN  
...             ...             ...      ...                     ...  
3813            0.0             6.0      1.0              2014-03-17  
3814            6.0             8.0      0.0                     NaN  
3815            0.0             1.0      0.0                     NaN  
3816            5.0             3.0      0.0              2014-01-20  
3817            2.0             8.0      0.0              2012-09-15  

[3818 rows x 17 columns]

In [7]:
train

tahun_kelahiran pendidikan status_pernikahan   pendapatan  \
0                1979    Sarjana   Rencana Menikah          NaN   
1                1950    Sarjana   Rencana Menikah   84063000.0   
2                1966    Sarjana           Menikah  127532564.0   
3                1961   Magister   Rencana Menikah  165579620.0   
4                1970    Sarjana   Rencana Menikah  117703159.0   
...               ...        ...               ...          ...   
3812             1955   Magister           Menikah   78199470.0   
3813             1947     Doktor   Rencana Menikah  109306000.0   
3814             1974   Magister           Menikah  104621000.0   
3815             1957        SMA   Rencana Menikah  110850000.0   
3816             1961    Sarjana           Menikah  130512000.0   

      jumlah_anak_balita  jumlah_anak_remaja  terakhir_belanja  belanja_buah  \
0                    0.0                 1.0               NaN       50575.0   
1                    NaN                 NaN              70.0        6069.0   
2                    0.0                 0.0              45.0      117611.0   
3                    0.0                 0.0              90.0      206346.0   
4                    1.0                 1.0              78.0       90563.0   
...                  ...                 ...               ...           ...   
3812                 0.0                 0.0              33.0        6069.0   
3813                 0.0                 1.0              44.0           0.0   
3814                 0.0                 2.0              68.0        2023.0   
3815                 1.0                 1.0              67.0       18207.0   
3816                 0.0                 0.0              70.0      329749.0   

      belanja_daging  belanja_ikan  belanja_kue  pembelian_diskon  \
0           260967.0       50575.0      20230.0               2.0   
1            44506.0       80920.0      20230.0               9.0   
2           265460.0       96341.0     145573.0               1.0   
3          1613901.0       27725.0     125868.0               0.0   
4           311757.0       40358.0      33875.0               7.0   
...              ...           ...          ...               ...   
3812         25977.0        3856.0       5784.0               5.0   
3813         50575.0           NaN          0.0               3.0   
3814         62713.0        8092.0          0.0               7.0   
3815         70805.0       24276.0          NaN               4.0   
3816        600831.0      420784.0     161840.0               3.0   

      pembelian_web  pembelian_toko  keluhan tanggal_menjadi_anggota  
0               2.0             5.0      0.0              2014-05-05  
1               6.0             4.0      0.0              2013-03-17  
2               1.0             7.0      0.0                     NaN  
3               7.0             8.0      0.0                     NaN  
4               6.0             5.0      0.0                     NaN  
...             ...             ...      ...                     ...  
3812            1.0             0.0      0.0                     NaN  
3813            6.0             3.0      0.0              2014-06-09  
3814            5.0             7.0      0.0              2013-11-07  
3815            5.0             4.0      0.0              2013-06-30  
3816            7.0            15.0      0.0              2014-01-25  

[3817 rows x 16 columns]

In [3]:
test.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3818 entries, 0 to 3817
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       3818 non-null   int64  
 1   tahun_kelahiran          3818 non-null   int64  
 2   pendidikan               3614 non-null   object 
 3   status_pernikahan        3636 non-null   object 
 4   pendapatan               3615 non-null   float64
 5   jumlah_anak_balita       3609 non-null   float64
 6   jumlah_anak_remaja       3608 non-null   float64
 7   terakhir_belanja         3617 non-null   float64
 8   belanja_buah             3632 non-null   float64
 9   belanja_daging           3623 non-null   float64
 10  belanja_ikan             3622 non-null   float64
 11  belanja_kue              3631 non-null   float64
 12  pembelian_diskon         3639 non-null   float64
 13  pembelian_web            3638 non-null   float64
 14  pembelian_toko          

In [4]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3817 entries, 0 to 3816
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   tahun_kelahiran          3817 non-null   int64  
 1   pendidikan               3628 non-null   object 
 2   status_pernikahan        3605 non-null   object 
 3   pendapatan               3627 non-null   float64
 4   jumlah_anak_balita       3627 non-null   float64
 5   jumlah_anak_remaja       3613 non-null   float64
 6   terakhir_belanja         3645 non-null   float64
 7   belanja_buah             3636 non-null   float64
 8   belanja_daging           3639 non-null   float64
 9   belanja_ikan             3624 non-null   float64
 10  belanja_kue              3603 non-null   float64
 11  pembelian_diskon         3639 non-null   float64
 12  pembelian_web            3652 non-null   float64
 13  pembelian_toko           3648 non-null   float64
 14  keluhan                 

In [ ]:
print("Statistik deskriptif data test:")
test.describe().T

Statistik deskriptif data test:


count          mean           std        min  \
ID                  3818.0  3.841507e+03  2.198401e+03        2.0   
tahun_kelahiran     3818.0  1.967790e+03  1.180857e+01     1892.0   
pendapatan          3615.0  1.163249e+08  5.105122e+07  2998899.0   
jumlah_anak_balita  3609.0  2.829039e-01  4.650046e-01        0.0   
jumlah_anak_remaja  3608.0  3.553215e-01  4.968651e-01        0.0   
terakhir_belanja    3617.0  4.718137e+01  2.694029e+01        0.0   
belanja_buah        3632.0  6.123408e+04  7.664759e+04        0.0   
belanja_daging      3623.0  4.528484e+05  5.123206e+05        0.0   
belanja_ikan        3622.0  8.533861e+04  1.037895e+05        0.0   
belanja_kue         3631.0  6.380151e+04  7.842854e+04        0.0   
pembelian_diskon    3639.0  2.101676e+00  2.115571e+00        0.0   
pembelian_web       3638.0  4.465091e+00  2.935615e+00        0.0   
pembelian_toko      3632.0  5.835628e+00  3.288349e+00        0.0   
keluhan             3625.0  2.758621e-03  5.245731e-02        0.0   

                            25%          50%           75%           max  
ID                      1950.75       3872.5  5.770750e+03  7.634000e+03  
tahun_kelahiran         1959.00       1968.0  1.976000e+03  2.009000e+03  
pendapatan          81681624.00  117732079.0  1.506877e+08  1.305740e+09  
jumlah_anak_balita         0.00          0.0  1.000000e+00  2.000000e+00  
jumlah_anak_remaja         0.00          0.0  1.000000e+00  2.000000e+00  
terakhir_belanja          25.00         46.0  6.800000e+01  1.330000e+02  
belanja_buah            7497.50      29525.0  8.677675e+04  4.086460e+05  
belanja_daging         54621.00     232753.0  7.217960e+05  3.271191e+06  
belanja_ikan           10115.00      40025.5  1.296568e+05  5.749320e+05  
belanja_kue             8092.00      29251.0  9.140700e+04  5.320490e+05  
pembelian_diskon           0.00          2.0  3.000000e+00  1.700000e+01  
pembelian_web              2.00          4.0  6.000000e+00  2.700000e+01  
pembelian_toko             3.00          5.0  8.000000e+00  1.700000e+01  
keluhan                    0.00          0.0  0.000000e+00  1.000000e+00

In [ ]:
print("\nStatistik deskriptif data train:")
train.describe().T


Statistik deskriptif data train:


count          mean           std        min         25%  \
tahun_kelahiran     3817.0  1.967824e+03  1.176813e+01     1899.0      1959.0   
pendapatan          3627.0  1.144832e+08  4.346042e+07  5073000.0  81125122.0   
jumlah_anak_balita  3627.0  2.930797e-01  4.730627e-01        0.0         0.0   
jumlah_anak_remaja  3613.0  3.537227e-01  4.930139e-01        0.0         0.0   
terakhir_belanja    3645.0  4.723155e+01  2.706851e+01        0.0        25.0   
belanja_buah        3636.0  5.980424e+04  7.402498e+04        0.0      7907.0   
belanja_daging      3639.0  4.385748e+05  5.120427e+05        0.0     49479.5   
belanja_ikan        3624.0  8.142900e+04  9.997623e+04        0.0     10115.0   
belanja_kue         3603.0  6.337797e+04  7.943546e+04        0.0      7947.0   
pembelian_diskon    3639.0  2.125584e+00  2.100133e+00        0.0         0.0   
pembelian_web       3652.0  4.436473e+00  3.002522e+00        0.0         2.0   
pembelian_toko      3648.0  5.767818e+00  3.210738e+00        0.0         3.0   
keluhan             3621.0  4.971002e-03  7.033959e-02        0.0         0.0   

                            50%          75%          max  
tahun_kelahiran          1968.0       1976.0       2000.0  
pendapatan          115621394.0  150495956.0  332884000.0  
jumlah_anak_balita          0.0          1.0          2.0  
jumlah_anak_remaja          0.0          1.0          2.0  
terakhir_belanja           47.0         69.0        128.0  
belanja_buah            26456.0      86162.0     396508.0  
belanja_daging         221993.0     686355.5    3489675.0  
belanja_ikan            36054.5     121380.0     621600.0  
belanja_kue             27795.0      89502.5     542164.0  
pembelian_diskon            2.0          3.0         20.0  
pembelian_web               4.0          7.0         30.0  
pembelian_toko              5.0          8.0         17.0  
keluhan                     0.0          0.0          1.0

Berdasarkan info dan statistik deskriptif dari data train dan data test, dapat diidentifikasi beberapa perbedaan:
- Jumlah data: Jumlah data yang dimiliki oleh data test yaitu 3818 yang mana sedikit lebih banyak dibanding data train yaitu 3817.
- Variabel ID hanya terdapat pada data test, data train tidak memiliki
- Pada variabel pendapatan pada data test memiliki nilai mean, max, dan standar deviasi yang lebih tinggi daripada data train.
- Pada variabel pembelian_diskon, data test memiliki nilai maksimum nya 17 yang mana lebih rendah dari nilai maksimum data train yaitu 20.
- Pada variabel terakhir_belanja, data test memiliki nilai maksimum nya 133 yang mana lebih tinggi dari nilai maksimumnya data train yaitu 128
- Pada varibel tahun_kelahiran, data test memiliki range yang lebih panjang yaitu 1829-2009 dibanding range data train yaitu 1899-2000

Dari beberapa variabel diatas sudah dapat dilihat bahwa terdapat perbedaan karakteristik antara data train dengan data test.